core/src/main/kotlin/fr/sncf/osrd/api/api_v2/pathfinding/PathfindingBlocksEndpointV2.kt

```diff
@@ -0,0 +1,279 @@
+    val res = mutableListOf<PathfindingConstraint<Block>>()
+    if (!rollingStockIsThermal) {
+        res.add(
+            ElectrificationConstraints(
+                infra.blockInfra,
+                infra.rawInfra,
+                rollingStockSupportedElectrification
+            )
+        )
+    }
+    res.add(LoadingGaugeConstraints(infra.blockInfra, infra.rawInfra, rollingStockLoadingGauge))
+    val sigSystemIds =
+        rollingStockSupportedSignalingSystems.map {
+            infra.signalingSimulator.sigModuleManager.findSignalingSystem(it)
+        }
+    res.add(SignalingSystemConstraints(infra.blockInfra, listOf(sigSystemIds)))
+    return res
+}
+
+@Throws(OSRDError::class)
+private fun computePaths(
+    infra: FullInfra,
+    waypoints: ArrayList<Collection<PathfindingEdgeLocationId<Block>>>,
+    constraints: List<PathfindingConstraint<Block>>,
+    remainingDistanceEstimators: List<AStarHeuristicId<Block>>,
+    timeout: Double?
+): PathfindingResultId<Block> {
+    val pathFound =
+        Pathfinding(GraphAdapter(infra.blockInfra, infra.rawInfra))
+            .setTimeout(timeout)
+            .setEdgeToLength { block: BlockId -> infra.blockInfra.getBlockLength(block) }
+            .setRemainingDistanceEstimator(remainingDistanceEstimators)
+            .addBlockedRangeOnEdges(constraints)
+            .runPathfinding(waypoints)
+    if (pathFound != null) {
+        return pathFound
+    }
+
+    // Handling errors
+    // Check if pathfinding failed due to constraints
+    val possiblePathWithoutErrorNoConstraints =
+        Pathfinding(GraphAdapter(infra.blockInfra, infra.rawInfra))
+            .setEdgeToLength { block -> infra.blockInfra.getBlockLength(block) }
+            .setRemainingDistanceEstimator(remainingDistanceEstimators)
+            .runPathfinding(waypoints)
+    if (possiblePathWithoutErrorNoConstraints != null) {
+        for (currentConstraint in constraints) {
+            val res =
+                Pathfinding(GraphAdapter(infra.blockInfra, infra.rawInfra))
+                    .setTimeout(timeout)
+                    .setEdgeToLength { block: BlockId -> infra.blockInfra.getBlockLength(block) }
+                    .addBlockedRangeOnEdges(currentConstraint)
+                    .setRemainingDistanceEstimator(remainingDistanceEstimators)
+                    .runPathfinding(waypoints)
+            if (res == null) {
+                // This way of handling it is suboptimal, but it should be reworked soon
+                when (currentConstraint::class.java) {
+                    ElectrificationConstraints::class.java -> {
+                        throw NoPathFoundException(
+                            IncompatibleElectrification(
+                                listOf(),
+                                listOf(),
+                                listOf(),
+                                Length(0.meters),
+                                listOf(),
+                            )
```

Comment:
I'm assuming that not having any detail is intended

Reply:
It's planned as a different feature, planned for the next PI if I recall correctly. For now we mostly keep the same behavior as before.